tests/metrics/classification/test_helpers.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+from pytorch_lightning.metrics.classification.helpers import DataType
+
+@pytest.mark.parametrize('s, data_type',
+    [
+        ("binary", DataType.BINARY), 
+        ("multilabel", DataType.MULTILABEL),
+        ("multiclass", DataType.MULTICLASS),
+        ("mdmc", DataType.MDMC),
+    ],
+)
+def test_data_type_equality(s, data_type):
+    assert s == data_type
+    assert data_type == s
+    assert s.upper() == data_type
+    assert data_type == s.upper()
+    assert DataType.from_str(s) == data_type
+    assert DataType.from_str(s.upper()) == data_type
```

Comment:
I think there is no need for this. It should already be tested in `LightningEnum` since this is no feature you added here...